ipc/src/platform/windows/channel/async_channel.rs

```diff
@@ -0,0 +1,144 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use crate::platform::metadata::ProcessHandle;
+use crate::platform::Channel;
+use pin_project::pin_project;
+use std::fmt::Debug;
+use std::os::windows::io::AsRawHandle;
+use std::{
+    io,
+    sync::{Arc, Mutex},
+    task::Poll,
+};
+use tokio::io::{AsyncRead, AsyncWrite};
+use tokio::net::windows::named_pipe::{NamedPipeClient, NamedPipeServer};
+use winapi::shared::wtypesbase::ULONG;
+use winapi::um::winbase::{GetNamedPipeClientProcessId, GetNamedPipeServerProcessId};
+use winapi::um::winnt::HANDLE;
+
+use super::ChannelMetadata;
+
+#[derive(Debug)]
+#[pin_project(project = NamedPipeProject)]
```

Comment:
I think it would be good to add some comments and/or links to docs explaining projections and structural pinning for ppl unfamiliar with the code base and Rust.